include/rocksdb/async/coro_promise.h

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+// Copyright (c) 2011 The LevelDB Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file. See the AUTHORS file for names of contributors.
+//
+// Experimental coroutines support to explore async implementations.
+//
+// For detailed information see: 
+// http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2017/n4649.pdf
+//
+// Quick cookbooks
+// https://youtu.be/ZTqHjjm86Bw
+// https://youtu.be/8C8NnE1Dg4A?t=2414
+// 
+
+#pragma once
+
+#ifndef STORAGE_ROCKSDB_INCLUDE_CORO_PROMISE_H_
+#define STORAGE_ROCKSDB_INCLUDE_CORO_PROMISE_H_
+
+#if !(defined ROCKSDB_COROUTINES)
```

Comment:
Compiler flags are defined when compiling RocksDB. Now we require them to define it when building their application depends on RocksDB. I'm not sure the interface would work well in the long term.

I know it's painful but this may have to be replaced with actual flags that can distinguish compilers.